discovery-provider/src/queries/get_trending_tracks.py

```diff
@@ -146,9 +168,37 @@ def generate_unpopulated_trending_from_mat_views(
             TrackTrendingScore.genre == genre
         )
 
+    # If usdc_purchase_only is true, then filter out track ids belonging to
+    # non-USDC purchase tracks before applying the limit.
+    if usdc_purchase_only:
+        trending_track_ids_subquery = trending_track_ids_query.subquery()
+        trending_track_ids = (
+            session.query(
+                trending_track_ids_subquery.c.track_id,
+                trending_track_ids_subquery.c.score,
+                Track.track_id,
+            )
+            .join(
+                trending_track_ids_subquery,
+                Track.track_id == trending_track_ids_subquery.c.track_id,
+            )
+            .filter(
+                Track.is_current == True,
+                Track.is_delete == False,
+                Track.stem_of == None,
+                Track.is_premium == True,
+                text("CAST(premium_conditions AS TEXT) LIKE '%usdc_purchase%'"),
```

Comment:
how is performance against prod db?

Reply:
`select * from tracks where is_current and not is_delete and is_premium and stem_of is null and CAST(premium_conditions AS TEXT) LIKE '%usdc_purchase%';` instantly gave me the results of one record, which was added today. few runs took an average of about 65ms